### PR TITLE
 Support lookahead with scalars in ebpf 

### LINKED ISSUE #2532: EBPF backend does not support non-header for lookahead
In `p4c/testdata/p4_16_samples/cal-ebpf.p4`, I see lookahead is supported by the ebpf backend.  However, I need a bit-field returned from lookahead to be also supported because, for example, I have to parse inner type of a GTP tunnel to be ipv4 or ipv6.  Thus, I am forced to use this lookahead which fails compiling with p4c-ebpf.

```
state parse_gtp_payload {
        bit<4> version = pkt.lookahead<bit<4>>();
        transition select(version) {
            4w4: parse_inner_ipv4;
            4w6: parse_inner_ipv6;
        }
    } 
```

To reproduce the issue, I changed switch_epbf.p4 for a contrived change and the problem is reproduced.  See log below.

```
$ ./p4c-ebpf -o tmp.c ../testdata/p4_16_samples/switch_ebpf.p4
../testdata/p4_16_samples/switch_ebpf.p4(43): [--Werror=legacy] error: Cannot extract to a non-struct type version
 bit<4> version = p.lookahead<bit<4>>();
        ^^^^^^^
```

P4 program is attached.

[switch1_ebpf_p4.txt.txt](https://github.com/p4lang/p4c/files/5192164/switch1_ebpf_p4.txt.txt)

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
Fixes #2532

### CODE REVIEW

**Comment by mbudiu-vmw** [comment]
I think I have lost a commit here, the actual intended fix is not here.

**Comment by mbudiu-vmw** [comment]
I was pretty sure I had committed this code already, but I could not find it anywhere. This PR should now be ready for review.
I would appreciate a review, since this also fixes the broken build for p4c-xdp, which was broken by a prior commit which I merged without enough testing.

**Comment by mbudiu-vmw** [comment]
@hanw: perhaps you can review this quickly; it is a small change.
The ebpf back-end is now broken until we merge this.

File: backends/ebpf/targets/xdp_target.py
Comment (rst0git) [inline backends/ebpf/targets/xdp_target.py:1]: The path in this symlink should be relative.

Comment (mbudiu-vmw) [reply backends/ebpf/targets/xdp_target.py:1]: This file should not be here at all, in fact.

File: testdata/p4_16_samples_outputs/issue2543-2-midend.p4
Comment (rst0git) [inline testdata/p4_16_samples_outputs/issue2543-2-midend.p4:15]: Was this line added intentionally?

Comment (mbudiu-vmw) [reply testdata/p4_16_samples_outputs/issue2543-2-midend.p4:15]: These are automatically generated files

**Review by liujed-intel — APPROVED** [review]
Approving to unblock, and to honour a [contract](https://github.com/p4lang/p4c/pull/2600#issuecomment-721477394). I'm not entirely familiar with the EBPF back end, but these changes look reasonable.

**Review by rst0git — APPROVED** [review]

**Pull request merged**
